compiler/main/driver.cpp

```diff
@@ -765,6 +784,84 @@ static void handleIncDir(const ArgumentDescription* desc, const char* arg_unused
   addIncInfo(incFilename);
 }
 
+static int invokeChplWithArgs(int argc, char* argv[],
+                              const std::vector<std::string>& additionalArgs,
+                              const char* description) {
+  // invoke the compiler again with arguments forwarded
+  assert(!additionalArgs.empty() &&
+         "expected additional flags to be specified for chpl invocation");
+  std::vector<std::string> commandVec;
+  for (int i = 0; i < argc; i++) {
+    commandVec.emplace_back(argv[i]);
+    if (i == 0) {
+      commandVec.insert(commandVec.end(), additionalArgs.begin(),
+                        additionalArgs.end());
+    }
+  }
+
+  return mysystem(commandVec, description,
+                  /* ignoreStatus */ true, /* quiet */ false);
+}
+
+static int runCompilation(int argc, char* argv[]);
+static int runMakeBinary(int argc, char* argv[]);
+
+// Use 'chpl' executable as a compiler-driver, re-invoking itself with flags
+// that trigger components of the actual compilation work to be performed.
+// After all phases are complete, or upon failure, the driver exits.
+static void runAsCompilerDriver(int argc, char* argv[]) {
+  int status = 0;
+
+  // initialize resources that need to be carried over between invocations
+  ensureTmpDirExists();
+  saveCompileCommand();
+
+  // invoke compilation
+  if ((status = runCompilation(argc, argv)) != 0) {
+    clean_exit(status);
+  }
+
+  // Skip make binary if the compile command does not require it, or if we are
+  // running in a debugger for the compilation phase only.
+  bool skipDueToDebug = (fRungdb || fRunlldb) &&
+    (driverDebugCompilation && !driverDebugMakeBinary);
+  if (skipDueToDebug) {
+    USR_WARN("Skipping make-binary phase due to running only compilation phase "
+        "in debugger; change this with --driver-debug-phase if desired");
```

Comment:
This confused me. Is the intention to not do the make binary phase if we're running inside the debugger or do we just run the make binary outside of the debugger?

Reply:
The intention is to not run the make-binary phase ("phase 2" now) at all if we only ran the compilation phase ("phase 1") in the debugger. I wrote about this incorrectly in the OP and have since fixed it to reflect my intention and what the code does.